File: packages/typia/src/transformers/features/http/HttpValidateHeadersTransformer.ts
import { HttpValidateHeadersProgrammer } from "../../../programmers/http/HttpValidateHeadersProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace HttpValidateHeadersTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "http.validateHeaders",
      write: HttpValidateHeadersProgrammer.write,
    });
}

File: packages/transform/src/features/http/HttpValidateHeadersTransformer.go
package http

import (
	corehttp "github.com/samchon/typia/packages/core/src/programmers/http"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var HttpValidateHeadersTransformer = httpValidateHeadersTransformerNamespace{}

type httpValidateHeadersTransformerNamespace struct{}

func (httpValidateHeadersTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return corehttp.HttpValidateHeadersProgrammer.Write(call.TypeText)
}
